diff --git a/.gitignore b/.gitignore
index 48bbc7e0..e8b338f8 100644
--- a/.gitignore
+++ b/.gitignore
@@ -7,3 +7,4 @@ Cargo.lock
 /.idea
 /.vscode
 /benches/target
+*.sw*
diff --git a/Cargo.toml b/Cargo.toml
index 7890580e..ce0f55f8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -86,7 +86,7 @@ anyhow = "1.0"
 rand = "0.8.0"
 ron = "0.6.2"
 serde = {version = "1", features = ["derive"]}
-openxr = { version = "0.14", features = ["loaded"], default-features = false } # FIXME optional based on feature
+openxr = { path = "../openxrs/openxr", features = ["loaded"], default-features = false } # FIXME optional based on feature
 wgpu = "0.7" # FIXME optional based on feature
 image = "0.23.13"
 log = "0.4.14"
@@ -429,3 +429,16 @@ target_sdk_version = 29
 
 [patch.crates-io]
 wgpu = { path = "../wgpu-rs" }
+
+[patch."https://github.com/gfx-rs/gfx"]
+gfx-hal = { path = "../gfx/src/hal" }
+gfx-backend-empty = { path = "../gfx/src/backend/empty" }
+gfx-backend-vulkan = { path = "../gfx/src/backend/vulkan" }
+gfx-backend-gl = { path = "../gfx/src/backend/gl" }
+gfx-backend-dx12 = { path = "../gfx/src/backend/dx12" }
+gfx-backend-dx11 = { path = "../gfx/src/backend/dx11" }
+gfx-backend-metal = { path = "../gfx/src/backend/metal" }
+
+[patch."https://github.com/gfx-rs/wgpu"]
+wgpu-types = { version = "0.7", path = "../wgpu/wgpu-types" }
+wgpu-core = { version = "0.7", path = "../wgpu/wgpu-core" }
