diff --git a/src/backend/vulkan/Cargo.toml b/src/backend/vulkan/Cargo.toml
index 7705f33d..373ee5eb 100644
--- a/src/backend/vulkan/Cargo.toml
+++ b/src/backend/vulkan/Cargo.toml
@@ -31,7 +31,7 @@ parking_lot = "0.11"
 smallvec = "1.0"
 raw-window-handle = "0.3"
 inplace_it = "0.3.3"
-openxr = { version = "0.14", features = ["loaded"], default-features = false, optional = true }
+openxr = { path = "../../../../openxrs/openxr", features = ["loaded"], default-features = false, optional = true }
 once_cell = { version = "1.5.2", optional = true }
 bitflags = "1" # FIXME: not required? features to wgt
 
diff --git a/src/hal/Cargo.toml b/src/hal/Cargo.toml
index fc60870c..537644a9 100644
--- a/src/hal/Cargo.toml
+++ b/src/hal/Cargo.toml
@@ -25,7 +25,7 @@ naga = { git = "https://github.com/gfx-rs/naga", tag = "gfx-9" }
 raw-window-handle = "0.3"
 serde = { version = "1", features = ["serde_derive"], optional = true }
 thiserror = "1"
-openxr = { version = "0.14", features = ["loaded"], default-features = false, optional = true }
+openxr = { path = "../../../openxrs/openxr", features = ["loaded"], default-features = false, optional = true }
 
 [dev-dependencies]
 gfx-backend-empty = { path = "../backend/empty", version = "0.7" }
