diff --git a/Cargo.toml b/Cargo.toml
index d329d43..dab0754 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,8 +7,8 @@ members = [
 ]
 
 [patch."https://github.com/gfx-rs/gfx"]
-#hal = { package = "gfx-hal", path = "../gfx/src/hal" }
-#gfx-backend-vulkan = { path = "../gfx/src/backend/vulkan", features = ["naga"] }
+hal = { package = "gfx-hal", path = "../gfx/src/hal" }
+gfx-backend-vulkan = { path = "../gfx/src/backend/vulkan", features = ["naga"] }
 #gfx-backend-metal = { path = "../gfx/src/backend/metal", features = ["naga"] }
 #gfx-backend-gl = { path = "../gfx/src/backend/gl", features = ["naga"] }
 #gfx-backend-dx12 = { path = "../gfx/src/backend/dx12" }
diff --git a/wgpu-core/Cargo.toml b/wgpu-core/Cargo.toml
index 8cf11d4..8b405bf 100644
--- a/wgpu-core/Cargo.toml
+++ b/wgpu-core/Cargo.toml
@@ -46,24 +46,26 @@ thiserror = "1"
 gpu-alloc = { version = "0.3", features = ["tracing"] }
 gpu-descriptor = { version = "0.1", features = ["tracing"] }
 
-hal = { package = "gfx-hal", git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7" }
-gfx-backend-empty = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7" }
+hal = { package = "gfx-hal", path = "../../gfx/src/hal" }
+gfx-backend-empty = { path = "../../gfx/src/backend/empty" }
+
+openxr = { path = "../../openxrs/openxr", features = ["loaded"], default-features = false, optional = true }
 
 [target.'cfg(all(not(target_arch = "wasm32"), all(unix, not(target_os = "ios"), not(target_os = "macos"))))'.dependencies]
-gfx-backend-vulkan = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7", features = ["naga"] }
-gfx-backend-gl = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7", features = ["naga"] }
+gfx-backend-vulkan = { path = "../../gfx/src/backend/vulkan", features = ["naga"] }
+gfx-backend-gl = { path = "../../gfx/src/backend/gl", features = ["naga"] }
 
 [target.'cfg(all(not(target_arch = "wasm32"), any(target_os = "ios", target_os = "macos")))'.dependencies]
-gfx-backend-metal = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7", features = ["naga"] }
-gfx-backend-vulkan = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7", optional = true }
+gfx-backend-metal = { path = "../../gfx/src/backend/metal", features = ["naga"] }
+gfx-backend-vulkan = { path = "../../gfx/src/backend/vulkan", optional = true }
 
 [target.'cfg(all(not(target_arch = "wasm32"), windows))'.dependencies]
-gfx-backend-dx12 = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7" }
-gfx-backend-dx11 = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7" }
-gfx-backend-vulkan = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7", features = ["naga"] }
+gfx-backend-dx12 = { path = "../../gfx/src/backend/dx12" }
+gfx-backend-dx11 = { path = "../../gfx/src/backend/dx11" }
+gfx-backend-vulkan = { path = "../../gfx/src/backend/vulkan", features = ["naga"] }
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-gfx-backend-gl = { git = "https://github.com/gfx-rs/gfx", rev = "d54cdcfac68711a91b55682c56da09f8e5b6f4e7", features = ["naga"] }
+gfx-backend-gl = { path = "../../gfx/src/backend/gl", features = ["naga"] }
 
 [dependencies.naga]
 git = "https://github.com/gfx-rs/naga"
diff --git a/wgpu-types/Cargo.toml b/wgpu-types/Cargo.toml
index 99f8c18..0cd3146 100644
--- a/wgpu-types/Cargo.toml
+++ b/wgpu-types/Cargo.toml
@@ -19,4 +19,4 @@ use-openxr = ["openxr"]
 [dependencies]
 bitflags = "1.0"
 serde = { version = "1.0", features = ["serde_derive"], optional = true }
-openxr = { version = "0.14", features = ["loaded"], default-features = false, optional = true }
+openxr = { path = "../../openxrs/openxr", features = ["loaded"], default-features = false, optional = true }
